bootstrap/023_simple_materializations.sql

```diff
@@ -93,3 +94,42 @@ begin
     call internal.refresh_warehouses();
     return 'success';
 end;
+
+CREATE TABLE IF NOT EXISTS internal_reporting_mv.warehouse_load_history (start_time timestamp_ltz,
+        end_time timestamp_ltz,
+        warehouse_name text,
+        avg_running NUMBER(38,2),
+        avg_queued_load NUMBER(38,2),
+        avg_queued_provisioning NUMBER(38,2),
+        avg_blocked NUMBER(38,2));
+create or replace view reporting.warehouse_load_history as select * from internal_reporting_mv.warehouse_load_history;
+
+
+CREATE OR REPLACE PROCEDURE internal.refresh_one_warehouse_load_history(warehouse_name varchar) RETURNS table(varchar) LANGUAGE SQL
+    AS
+begin
+    let dt timestamp := current_timestamp();
+
+    let input variant := null;
+    BEGIN
+        input := (select output from INTERNAL.TASK_WAREHOUSE_LOAD_EVENTS where success and warehouse_name = :warehouse_name order by run desc limit 1);
+        let oldest_running timestamp := 0::timestamp;
+
+        if (input is not null) then
+            oldest_running := coalesce(input:oldest_running::timestamp, 0::timestamp);
+        end if;
+
+        let stmt varchar := 'insert into internal_reporting_mv.warehouse_load_history select start_time, end_time, warehouse_name, avg_running, avg_queued_load, avg_queued_provisioning, avg_blocked from table(information_schema.warehouse_load_history(date_range_start => to_timestamp_ltz(\'{start_time}\'), date_range_end => to_timestamp_ltz(\'{end_time}\'), warehouse_name => \'{warehouse_name}\'))';
+        let start_time timestamp := (select greatest(:oldest_running, dateadd(day, -14, current_timestamp())));
+        let res resultset := (select tools.templatejs(:stmt,
+            {
+              'start_time': dateadd(hours, value, :start_time)::text,
+              'end_time': dateadd(hours, value+8, :start_time)::text
```

Comment:
need `warehouse_name` or does that somehow get substituted in?

Reply:
good shout